src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java

```diff
@@ -109,9 +111,15 @@
    * Optional task to run
    * @since 2.1
    */
-  private final String task;
+  private String task;
 
   @DataBoundConstructor
+  public SonarRunnerBuilder() {
+    // all fields are optional
+  }
+
+  // We're moving to using @DataBoundSetter instead and a much leaner @DataBoundConstructor
+  @Deprecated
   public SonarRunnerBuilder(String installationName, String sonarScannerName, String project, String properties, String javaOpts, String jdk, String task,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)